app/src/main/java/org/mozilla/fenix/share/listadapters/AppShareAdapter.kt

```diff
@@ -7,14 +7,19 @@ package org.mozilla.fenix.share.listadapters
 import android.graphics.drawable.Drawable
 import android.view.LayoutInflater
 import android.view.ViewGroup
-import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
 import org.mozilla.fenix.share.ShareToAppsInteractor
 import org.mozilla.fenix.share.viewholders.AppViewHolder
 
 class AppShareAdapter(
     private val interactor: ShareToAppsInteractor,
-    private val applications: MutableList<AppShareOption> = mutableListOf()
-) : RecyclerView.Adapter<AppViewHolder>() {
+    applications: List<AppShareOption> = emptyList()
+) : ListAdapter<AppShareOption, AppViewHolder>(DiffCallback()) {
+
+    init {
+        submitList(applications)
```

Comment:
I think we should just replace uses of the second constructor parameter with a call to `submitList`

Reply:
I guess it's a good idea.